stl/src/filesystem.cpp

```diff
@@ -899,6 +899,13 @@ namespace {
 
                 _Flags &= ~_Get_file_attributes_data;
             }
+
+            if (!_STD _Bitmask_includes_any(_Attributes, __std_fs_file_attr::_Reparse_point)
+                && _STD _Bitmask_includes_any(_Flags, __std_fs_stats_flags::_Reparse_tag)) {
+                // if reparse tag is requested by caller but the file is not a reparse point, set tag to _None
+                _Stats->_Reparse_point_tag = __std_fs_reparse_tag::_None;
+                _Flags &= ~__std_fs_stats_flags::_Reparse_tag;
+            }

```

Comment:
We have a check in this function below:
```C++
            // Calling GetFileInformationByHandleEx with FileAttributeTagInfo fails on FAT file system with
            // ERROR_INVALID_PARAMETER. We avoid calling this for non-reparse-points.
            if ((_Info.FileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) != 0u) {
```

I think we should drop that check after checking it here, as that one will always succeed then.

Reply:
Actually, if `_Follow_symlinks` is true `_Info.FileAttributes` can be attributes for another object and can differ from `_Data.dwFileAttributes`. So it is ok to perform this check.